roofit/roofit/src/RooBernstein.cxx

```diff
@@ -84,10 +84,20 @@ RooBernstein::RooBernstein(const RooBernstein& other, const char* name) :
 
 ////////////////////////////////////////////////////////////////////////////////
 
+void RooBernstein::selectNormalizationRange(const char* rangeName, Bool_t force)
+{
+  if (rangeName && (force || !_refRangeName.empty())) {
+     _refRangeName = static_cast<const TNamed*> (RooNameReg::instance().constPtr(rangeName));
+  }
+}
+
+////////////////////////////////////////////////////////////////////////////////
+
 Double_t RooBernstein::evaluate() const
 {
-  Double_t xmin = _x.min();
-  Double_t x = (_x - xmin) / (_x.max() - xmin); // rescale to [0,1]
+  const Double_t xmax = _x.max(_refRangeName?_refRangeName.c_str():0);
+  const Double_t xmin = _x.min(_refRangeName?_refRangeName.c_str():0);
```

Comment:
With new RooFit, you can do this in a single function call.
```suggestion
  double xmin, xmax;
  std::tie(xmin, xmax) = _x.getRange(_refRangeName.empty() ? nullptr : _refRangeName.c_str());
```
Same in the integral function.

Reply:
Hi Stephan,

    When I test this change, I get the following error whilst compiling root.

error: 'const RooRealProxy {aka const class RooProxy<RooAbsReal>}' has no member named 'getRange'; did you mean 'hasRange'?
   std::tie(xmin, xmax) = _x.getRange(_refRangeName.empty() ? nullptr : _refRangeName.c_str());